alternator/executor.cc

```diff
@@ -575,6 +585,28 @@ future<executor::request_return_type> executor::delete_table(client_state& clien
 
         std::move(m2.begin(), m2.end(), std::back_inserter(m));
 
+        // When deleting a table and its views, we need to remove this role's
+        // special permissions in those tables (undoing the "auto-grant" done
+        // by CreateTable). If we didn't do this, if a second role later
+        /// recreates a table with the same name, the first role would still
+        // have permissions over the new table.
+        // To make things more robust we just remove *all* permissions for
+        // the deleted table (CQL's drop_table_statement also does this).
+        // Unfortunately, there is an API mismatch between this code (which
+        // uses separate group0_guard and vector<mutation>) and the function
+        // revoke_all() which uses a combined "group0_batch" structure - so
+        // we need to do some ugly back-and-forth conversions between the pair
+        // to the group0_batch and back to the pair :-(
+        service::group0_batch mc(std::move(group0_guard));
+        mc.add_mutations(std::move(m));
+        co_await auth::revoke_all(*cs.get().get_auth_service(),
+            auth::make_data_resource(schema->ks_name(), schema->cf_name()), mc);
+        for (const view_ptr& v : tbl->views()) {
+            co_await auth::revoke_all(*cs.get().get_auth_service(),
+                auth::make_data_resource(v->ks_name(), v->cf_name()), mc);
+        }
+        std::tie(m, group0_guard) = co_await std::move(mc).extract();
```

Comment:
why not just `auto [m, group0_guard] = co_await ...`

Reply:
Because the syntax you recommend creates new variables - the one I used does not, and re-assigns the already existing variables. The syntax you recommend may work too, but I'm pretty sure the compiler won't like in the future (or might not even like it today) the fact I'm "shadowing" already existing variables - it's usually a sign of a bug and not something does deliberately.

As you can probably see from the comment, I really didn't like - or didn't understand - the group0_batch interface. It looks like our code world is split into code that uses that single group0_batch and code which uses a separate group0_guard and `vector<mutation>`, and it makes interfacing between those two different "worlds" really ugly, as I needed to do in this series twice. We should figure out a better way.

Reply:
Is this something you can shed some light onto @kbr-scylla ?

Reply:
I don't know. @nuivall is the author of `group0_batch`, he may have an opinion.